### ISSUE TITLE
When creating a machine of the same name as an existing host, the existing host is deleted

### ISSUE THREAD

**Issue by SvenDowideit** [docker/machine#249]
So I forgot i had a perfectly good and useful vm called `vmtest`, and when I re-ran the create on it by mistake, it deleted all my data!

seriously, this would be something to error out on, not to destroy something.

also - why is it re-downloading the iso, aren't we sharing?

```
mini:machine sven$ ./machine create --driver=vmwarefusion vmtest
ERRO[0000] Error creating host: Host "vmtest" already exists 
WARN[0000] Removing created machine. You can run machine with the --debug flag to avoid this. 
WARN[0002] You will want to check the provider to make sure the machine and associated resources were properly removed. 
FATA[0002] Error creating machine                       
mini:machine sven$ ./machine create --driver=vmwarefusion vmtest
INFO[0000] Downloading boot2docker...                   
```

**Comment by ehazlett** [comment]
The fusion driver has to use it's own build because of the vmtools.  As part of #217 we should perhaps version these ISOs by driver (currently vbox/fusion).

And +1 to @sthulb -- we should dupe check as early as possible.

**Comment by bfirsh** [comment]
Ouch. We should have a test for this. A good argument for having a set of integration tests for commands.

**Comment by ehazlett** [comment]
This has been fixed as of #267 and .  Closing.
